Add Python version

miekof opened the pull request:
This is a Python version of the same test.

miekof committed:
Add Python version

eabedrapo commented on python/HelloWorldTest.py:
```diff
@@ -0,0 +1,76 @@
+import re
+
+
+class HelloWorldTests (unittest.TestCase):
```
Unnecessary space between Class name and parenthesis.

miekof replied:
Removed

eabedrapo commented on python/HelloWorldTest.py:
```diff
@@ -0,0 +1,76 @@
+        # Verify obtained text is "Success"
+        self.assertEqual(success_text, "Success", "Message should be Success")
+
+    def earDown(self):
```
Shouldn't be "tearDown"?

miekof replied:
Yes, it should be.

eabedrapo commented on python/HelloWorldTest.py:
```diff
@@ -0,0 +1,76 @@
+
+    def test_verify_captcha(self):
+        # Go to the site
+        self.driver.get("https://www.ultimateqa.com/filling-out-forms/")
```
`self.driver` is used in many places across this test, maybe it would be worth to change it to get an alias this way:
`driver = self.driver`

Then you just use `driver.get()`, `driver.find_element()` and so on and so forth.

miekof replied:
updated.

eabedrapo reviewed (changes requested):
I think this script needs a few minor changes.

miekof committed:
Fix based on review

* Add driver=self.driver
* Fix a typo

Not a part of review:
* Remove setting the path to chromedriver.
If you follow the instruction in Wiki, no need
to set it. webdrivers must be in $PATH or %PATH%
variable.

eabedrapo commented on python/HelloWorldTest.py:
```diff
@@ -22,52 +19,48 @@ class HelloWorldTests (unittest.TestCase):
-        driver_path = r"drivers"
-        driver_name = r"chromedriver.exe" if _platform == "Windows" else r"chromedriver"
-        webdriver_path = os.path.join(user_home, driver_path, driver_name)
         chrome_options: Options = webdriver.ChromeOptions()
```
Just wondering, why do you need to set all these options for such a simple script? Shouldn't we keep these scripts simple as it's just intended to get beginners onboard with Selenium?

miekof replied:
This is already removed from the latest version.

miekof replied:
This is actually most problematic area for beginners to start running Selenium tests. (My hope is to prevent as much as possible.) Maybe We can provide a check list Selenium test can't find webdrivers.

eabedrapo commented on python/HelloWorldTest.py:
```diff
@@ -1,17 +1,14 @@
 
-class HelloWorldTests (unittest.TestCase):
+class HelloWorldTests(unittest.TestCase):
     TEN_SECOND = 10
```
TEN_SECONDS should be plural.

miekof replied:
Updated

eabedrapo commented on python/HelloWorldTest.py:
```diff
@@ -22,52 +19,48 @@ class HelloWorldTests (unittest.TestCase):
 
         # Find right input field
-        right_name_input_element = self.driver.find_element(By.ID, HelloWorldTests.RIGHT_NAME_INPUT_ID)
+        right_name_input_element = driver.find_element(By.ID, HelloWorldTests.RIGHT_NAME_INPUT_ID)
```
Although `HelloWorldTests.RIGHT_NAME_INPUT_ID` works, is better if you use `self.RIGHT_NAME_INPUT_ID` given the first way works similar to a "static" variable in Java while the second is a way to pass an instance variable.

Using the first way to refer instance variables might get undesired consequences, as if you extend this class and redefine the values for the variables listed at the top of the class the new values will never get used as the test is referring explicitly the variables defined in the mother class.

Also, using `self` makes all lines shorter and easier to read, don't you think? :wink:

miekof replied:
A good point. Python doesn't have `final` the same way Java does. Updated as suggested.

eabedrapo reviewed (commented):
Just a few more comments... feel free to disagree as this test was just a boilerplate and maybe I'm just overthinking too much, so if you prefer leaving the code as it is, it's OK by me 😉

miekof committed:
fixup! Fix based on review

eabedrapo commented:
LGTM.
